src/org/omegat/core/team2/impl/GITCredentialsProvider.java

```diff
@@ -80,36 +79,83 @@
 public class GITCredentialsProvider extends CredentialsProvider {
 
     static {
-        // Set up ssh-agent support
         JschConfigSessionFactory sessionFactory = new JschConfigSessionFactory() {
 
             @Override
             protected void configure(OpenSshConfig.Host host, Session session) {
                 session.setConfig("StrictHostKeyChecking", "true");
             }
 
+            /**
+             * Add ssh-agent support configuration to default JSch instance.
+             * @param jsch
+             */
             @Override
-            protected JSch createDefaultJSch(FS fs) throws JSchException {
-                Connector con = null;
+            protected void configureJSch(final JSch jsch) {
+                super.configureJSch(jsch);
                 try {
-                    if (SSHAgentConnector.isConnectorAvailable()) {
-                        USocketFactory usf = new JNAUSocketFactory();
-                        con = new SSHAgentConnector(usf);
-                    } else {
-                        ConnectorFactory cf = ConnectorFactory.getDefault();
-                        con = cf.createConnector();
-                    }
-                } catch (AgentProxyException e) {
-                    Log.log(e);
-                }
-                JSch jsch = super.createDefaultJSch(fs);
-                if (con != null) {
+                    // Use ssh-agent connector class provided by forked JSch project.
+                    IdentityRepository irepo = new AgentIdentityRepository(new SSHAgentConnector());
                     JSch.setConfig("PreferredAuthentications", "publickey");
-                    IdentityRepository irepo = new RemoteIdentityRepository(con);
                     jsch.setIdentityRepository(irepo);
+                } catch (AgentProxyException e) {
+                    Log.log(e);
                 }
+            }
+
+            /**
+             * Default JSch instance creator.
+             *
+             * default key names; identity, id_rsa, id_ecdsa, and id_ed25519.
+             * It solves BUG#1075, and realizes RFE#1598
+             */
+            @Override
+            protected JSch createDefaultJSch(FS fs) throws JSchException {
+                // Note: it should not call super.createDefaultJSch().
+                // we create instance from scratch to avoid
+                // the forked JSch and JGit combination issue that
+                // lead NPE.
+                JSch jsch = new JSch();
+                configureJSch(jsch);
+                knownHosts(jsch, fs);
+                identities(jsch, fs);
```

Comment:
There is a copy from parent class to avoid the JGit bug.
It may be better to have an intermediate class to proxy `JschConfigSessionFactory` and change behavior by back porting of the fix. 
I'd like to leave only a required code when next JGit version, 5.13.x,  is released with the fix.

When the fix released, we can remove intermediate patching class.

Reply:
done